python/triqs_cthyb/solver.py

```diff
@@ -145,11 +148,39 @@ def solve(self, **params_kw):
         # (only supported for G_tau, to permit compatibility with dft_tools)
         if perform_post_proc and (self.last_solve_parameters["measure_G_tau"] == True):
 
+            if self.last_solve_parameters["measure_density_matrix"]:
+                # we have the density matrix, so we will compute the high frequency
+                # moments and orbital occupations
+
+                self.orbital_occupations = orbital_occupations(self.density_matrix,
+                                                               self.gf_struct,
+                                                               self.h_loc_diagonalization
+                                                               )
+
+                h_int = self.last_solve_parameters['h_int']
+                self.Sigma_moments = sigma_high_frequency_moments(self.density_matrix,
+                                                 self.h_loc_diagonalization,
+                                                 self.gf_struct,
+                                                 h_int
+                                                 )
+
+                self.Sigma_Hartree = {bl: sigma_bl[0] for bl, sigma_bl in self.Sigma_moments.items()}
+
+                self.G_moments = green_high_frequency_moments(self.density_matrix,
+                                                         self.h_loc_diagonalization,
+                                                         self.gf_struct,
+                                                         self.h_loc
+                                                         )
```

Comment:
To avoid confusion we should define all `tail` or `moments` objects starting from the `0th` order, i.e. constant part. Adjust `sigma_high_frequency_moments` and `green_high_frequency_moments` accordingly.